src/components/panel/panel.spec.js

```diff
@@ -2147,6 +2147,192 @@ describe('$mdPanel', function() {
     });
   });
 
+  describe('interceptor logic: ', function() {
+    var interceptorTypes;
+
+    beforeEach(function() {
+      interceptorTypes = $mdPanel.interceptorTypes;
+    });
+
+    it('should throw when registering an interceptor without a type', function() {
+      openPanel();
+
+      expect(function() {
+        panelRef.registerInterceptor();
+      }).toThrowError('MdPanel: Interceptor type must be a string, instead got undefined');
+    });
+
+    it('should throw when registering an interceptor without a callback', function() {
+      openPanel();
+
+      expect(function() {
+        panelRef.registerInterceptor(interceptorTypes.CLOSE);
+      }).toThrowError('MdPanel: Interceptor callback must be a function, instead got undefined');
+    });
+
+    it('should execute the registered interceptors', function() {
+      var obj = { callback: function() {} };
+
+      openPanel();
+      spyOn(obj, 'callback');
+
+      panelRef.registerInterceptor(interceptorTypes.CLOSE, obj.callback);
+      panelRef._callInterceptors(interceptorTypes.CLOSE);
+
+      expect(obj.callback).toHaveBeenCalledWith(panelRef);
+    });
+
+    it('should execute the interceptors in reverse order', function() {
```

Comment:
This was agreed upon with @ErinCoughlan. I think it's also the way Angular runs `$formatters` and `$parsers`.